src/middleware/TestingHandler.ts

```diff
@@ -0,0 +1,271 @@
+/**
+ * -------------------------------------------------------------------------------------------
+ * Copyright (c) Microsoft Corporation.  All Rights Reserved.  Licensed under the MIT License.
+ * See License in the project root for license information.
+ * -------------------------------------------------------------------------------------------
+ */
+
+/**
+ * @module TestingHandler
+ */
+
+import { Context } from "../IContext";
+import { RequestMethod } from "../RequestMethod";
+
+import { Middleware } from "./IMiddleware";
+import { MiddlewareControl } from "./MiddlewareControl";
+import { generateUUID } from "./MiddlewareUtil";
+import { httpStatusCode, methodStatusCode } from "./options/TestingHandlerData";
+import { TestingHandlerOptions } from "./options/TestingHandlerOptions";
+import { TestingStrategy } from "./options/TestingStrategy";
+
+/**
+ * Class representing TestingHandler
+ * @class
+ * Class
+ * @implements Middleware
+ */
+export class TestingHandler implements Middleware {
+	/**
+	 * A member holding options to customize the handler behavior
+	 *
+	 * @private
+	 */
+	private options: TestingHandlerOptions;
+
+	/**
+	 * container for the manual map that has been written by the client
+	 *
+	 * @private
+	 */
+	private manualMap: Map<string, Map<string, number>>;
+
+	/**
+	 * dummy url that we are using, randomly generated everytime
+	 *
+	 * @private
+	 */
+	private redirectURL: string;
+
+	/**
+	 * @public
+	 * @constructor
+	 * To create an instance of Testing Handler
+	 * @param {TestingHandlerOptions} [options = new TestingHandlerOptions()] - The testing handler options instance
+	 * @param manualMap - The Map passed by user containing url-statusCode info
+	 * @returns An instance of Testing Handler
+	 */
+	public constructor(options: TestingHandlerOptions = new TestingHandlerOptions(), manualMap?: Map<string, Map<string, number>>) {
+		this.options = options;
+		this.manualMap = manualMap;
+		this.redirectURL = "https://graph.microsoft.com/" + generateUUID();
+	}
+
+	/**
+	 * Generates responseHeader
+	 * @private
+	 * @param {number} statusCode - the status code to be returned for the request
+	 * @param {string} requestID - request id
+	 * @param {string} requestDate - date of the request
+	 * @returns response Header
+	 */
+	private createResponseHeaders(statusCode: number, requestID: string, requestDate: string) {
+		const responseHeader: Headers = new Headers();
+
+		responseHeader.append("Cache-Control", "private");
```

Comment:
Set the Cache-Control header to no-store.  I don't see a reason why we would want any downstream cache to store this response and serve it up.  The response isn't coming over the wire so there is no performance gain to allow caching.